test/functional/tests/cache_ops/test_dynamic_cache_mode_switching.py

```diff
@@ -0,0 +1,321 @@
+        cache = casadm.start_cache(cache_dev, cache_mode_1, force=True)
+
+    with TestRun.step("Add core to the cache"):
+        core = cache.add_core(core_dev)
+
+    with TestRun.step("Run 'fio'"):
+        fio = (
+            Fio()
+            .create_command()
+            .io_engine(IoEngine.libaio)
+            .block_size(Size(4, Unit.KibiByte))
+            .read_write(io_mode)
+            .target(core.system_path)
+            .direct(1)
+            .run_time(timedelta(minutes=4))
+            .time_based()
+        )
+        fio_pid = fio.run_in_background()
+        time.sleep(5)
+
+    with TestRun.step(
+        f"Change cache mode to {cache_mode_2} with flush cache set to: {flush}"
+    ):
+        cache_mode_switch_output = cache.set_cache_mode(cache_mode_2, flush)
+
+    with TestRun.step(f"Check if cache mode has switched properly during IO"):
+        cache_mode_after_switch = cache.get_cache_mode()
+
+        if (
+            cache_mode_switch_output.exit_code == 0
+            and cache_mode_after_switch == cache_mode_2
+        ):
+            TestRun.LOGGER.info("Cache mode switched properly")
+        else:
+            TestRun.fail(
+                f"Cache mode did not switch properly! "
+                f"Exit code: {cache_mode_switch_output.exit_code}, "
+                f"Cache mode after switch: {cache_mode_after_switch}"
+            )
+
+    with TestRun.step("Stop 'fio'"):
+        TestRun.executor.kill_process(fio_pid)
+
+    with TestRun.step("Stop all caches"):
+        casadm.stop_all_caches()
+
+
+def storage_prepare():
+    cache_dev = TestRun.disks["cache"]
+    cache_dev.create_partitions([Size(2, Unit.GibiByte)])
+    core_dev = TestRun.disks["core"]
+    core_dev.create_partitions([Size(1, Unit.GibiByte)])
+    return cache_dev.partitions[0], core_dev.partitions[0]
+
+
+def check_cache_mode_operation(cache, core, io_size, cache_mode):
+    cache.reset_counters()
+
+    if cache_mode == CacheMode.WB or cache_mode == CacheMode.PT:
+        io_mode = ReadWrite.randrw
+        run_io_and_verify(cache, core, io_mode, io_size)
+
+    if cache_mode == CacheMode.WT:
+        io_mode = ReadWrite.randwrite
+        run_io_and_verify(cache, core, io_mode, io_size)
+
+    if cache_mode == CacheMode.WA:
+        io_mode = ReadWrite.randwrite
+        run_io_and_verify(cache, core, io_mode, io_size)
+        cache.reset_counters()
+        io_mode = ReadWrite.randread
+        run_io_and_verify(cache, core, io_mode, io_size)
+
+
+def run_io_and_verify(cache, core, io_mode, io_size):
+    fio_prepare(core, io_mode, io_size).run()
+    sync()
+    cache_mode = cache.get_cache_mode()
+    cache_stats = cache.get_statistics()
+    core_stats = core.get_statistics()
+
+    if cache_mode == CacheMode.WB:
+        if core_stats.block_stats.core.writes < core_stats.block_stats.exp_obj.writes:
+            TestRun.LOGGER.info("Write-Back cache mode works properly")
```

Comment:
General - do we need this kind of logs in tests? Don't we expect everything to work fine and log only if something is wrong? What do You think?

Reply:
Removed